apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/ShardMigrationExecutor.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright © 2018-2019 Apollo Foundation
+ */
+
+package com.apollocurrency.aplwallet.apl.core.shard;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import javax.enterprise.util.AnnotationLiteral;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+import com.apollocurrency.aplwallet.apl.core.db.ShardAddConstraintsSchemaVersion;
+import com.apollocurrency.aplwallet.apl.core.db.ShardInitTableSchemaVersion;
+import com.apollocurrency.aplwallet.apl.core.shard.commands.CopyDataCommand;
+import com.apollocurrency.aplwallet.apl.core.shard.commands.CreateShardSchemaCommand;
+import com.apollocurrency.aplwallet.apl.core.shard.commands.DataMigrateOperation;
+import com.apollocurrency.aplwallet.apl.core.shard.commands.DeleteCopiedDataCommand;
+import com.apollocurrency.aplwallet.apl.core.shard.commands.FinishShardingCommand;
+import com.apollocurrency.aplwallet.apl.core.shard.commands.ReLinkDataCommand;
+import com.apollocurrency.aplwallet.apl.core.shard.commands.UpdateSecondaryIndexCommand;
+import com.apollocurrency.aplwallet.apl.core.shard.observer.events.ShardChangeStateEvent;
+import com.apollocurrency.aplwallet.apl.core.shard.observer.events.ShardChangeStateEventBinding;
+import org.slf4j.Logger;
+
+/**
+ * Component for starting sharding process which contains several steps/states.
+ *
+ * @author yuriy.larin
+ */
+@Singleton
+public class ShardMigrationExecutor {
+    private static final Logger log = getLogger(ShardMigrationExecutor.class);
+    private final List<DataMigrateOperation> dataMigrateOperations = new LinkedList<>();
```

Comment:
What purpose of using LinkedList?

Reply:
Currently we have to execute steps is certain order, so we preserve steps order in Linked List.

Reply:
ArrayList also guarantee order preservation for elements.